internal/k8s/appprotectdos/app_protect_dos_resources.go

```diff
@@ -0,0 +1,151 @@
+package appprotectdos
+
+import (
+	"fmt"
+	"net"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var appProtectDosPolicyRequiredFields = [][]string{
+	{"spec", "mitigation_mode"},
+	{"spec", "automation_tools_detection"},
+	{"spec", "tls_fingerprint"},
+	{"spec", "signatures"},
+	{"spec", "bad_actors"},
+}
+
+var appProtectDosLogConfRequiredFields = [][]string{
+	{"spec", "content"},
+	{"spec", "filter"},
+}
+
+func validateRequiredFields(policy *unstructured.Unstructured, fieldsList [][]string) error {
+	for _, fields := range fieldsList {
+		field, found, err := unstructured.NestedMap(policy.Object, fields...)
+		if err != nil {
+			return fmt.Errorf("Error checking for required field %v: %w", field, err)
+		}
+		if !found {
+			return fmt.Errorf("Required field %v not found", field)
+		}
+	}
+	return nil
+}
+
+func validateRequiredFieldsNoCopy(policy *unstructured.Unstructured, fieldsList [][]string) error {
+	for _, fields := range fieldsList {
+		field, found, err := unstructured.NestedFieldNoCopy(policy.Object, fields...)
+		if err != nil {
+			return fmt.Errorf("Error checking for required field %v: %w", field, err)
+		}
+		if !found {
+			return fmt.Errorf("Required field %v not found", field)
+		}
+	}
+	return nil
+}
+
+// ValidateAppDosProtectLogConf validates LogConfiguration resource
+func ValidateAppProtectDosLogConf(logConf *unstructured.Unstructured) error {
```

Comment:
can we have a unit test for this function?

Reply:
Add unit test for ValidateAppProtectDosLogConf